analyses/malawi/notebooks/mwi_ecmwf_monthly_skill_dryspells.md

```diff
@@ -0,0 +1,827 @@
+    df_ds_res=df_ds_all.reset_index(drop=True)
+    a = [pd.date_range(*r, freq='D') for r in df_ds_res[ds_date_cols].values]
+    #join the daterange with the adm2, which create a column per date, then stack to have each adm2-date combination
+    df_ds_daterange=df_ds_res[[ds_adm_col]].join(pd.DataFrame(a)).set_index([ds_adm_col]).stack().droplevel(-1).reset_index()
+    df_ds_daterange.rename(columns={0:"date"},inplace=True)
+    #all dates in this dataframe had an observed dry spell, so add that information
+    df_ds_daterange["dryspell_obs"]=1
+    df_ds_daterange["date_month"]=df_ds_daterange.date.dt.to_period("M")
+    
+    #count the number of days within a year-month combination that had were part of a dry spell
+    df_ds_countmonth=df_ds_daterange.groupby([ds_adm_col,"date_month"],as_index=False).sum()
+    
+    df_ds_month=df_ds_countmonth[df_ds_countmonth.dryspell_obs>=min_ds_days_month]
+    
+    #TODO: this is not really generalizable
+    if shp_adm_col not in df_ds_month.columns:
+        df_adm2=gpd.read_file(adm2_bound_path)
+        df_ds_month=df_ds_month.merge(df_adm2[["ADM2_PCODE","ADM2_EN","ADM1_EN"]],left_on=ds_adm_col,right_on="ADM2_PCODE")
+        
+    df_ds_month_adm1=df_ds_month.groupby([shp_adm_col,"date_month"],as_index=False).count()
+    df_ds_month_adm1["dry_spell"]=np.where(df_ds_month_adm1.dryspell_obs>=min_adm_ds_month,1,0)
+    
+    return df_ds_month_adm1
+```
+
+### Load the forecast data
+And select the data of interest
+
+```python
+#read the ecmwf forecast per adm1 per date and concat all dates
+# the mwi_seasonal-monthly-single-levels_v5_interp*.csv contain results when interpolating the forecasts to be more granular, but results actually worsen with this
+all_files = glob.glob(os.path.join(ecmwf_country_data_processed_dir, "mwi_seasonal-monthly-single-levels_v5_2*.csv"))
+
+df_from_each_file = (pd.read_csv(f,parse_dates=["date"]) for f in all_files)
+df_for   = pd.concat(df_from_each_file, ignore_index=True)
+```
+
+```python
+#this should be the number of years*number of months
+print(len(all_files))
+```
+
+```python
+#number of years*months till may 2021
+21*12+5
+```
+
+```python
+#for now using mean cell as this requires one variable less to be set (else need to set percentage of cells)
+aggr_meth="mean_cell"
+#for earlier dates, the model included less members --> values for those members are nan --> remove those rows
+df_for = df_for[df_for[aggr_meth].notna()]
+df_for["season_approx"]=np.where(df_for.date.dt.month>=10,df_for.date.dt.year,df_for.date.dt.year-1)
+```
+
+```python
+sel_adm=["Southern"]
+sel_months=[1,2]
+sel_leadtime=[1,2,3,4,5,6]
+seas_years=range(2000,2020)
+
+adm_str="".join([a.lower() for a in sel_adm])
+month_str="".join([calendar.month_abbr[m].lower() for m in sel_months])
+lt_str="".join([str(l) for l in sel_leadtime])
+
+#for this analysis we are only interested in the southern region during the DecJanFeb period, since this is most sensitive to dry spells
```

Comment:
Should you add 12 to months since you also want December?

Reply:
Oops first it was decjanfeb, but now changed to only janfeb. Changed the text accordingly